test/e2e/tests/test_scenarios.py

```diff
@@ -37,6 +37,88 @@ class ParameterGroupHelper(helper.ResourceHelper):
+            assert expectations == state
+
+        for (key, value) in expectations.items():
+            # conditions are processed separately
+            if key == "conditions":
+                continue
+            # tags are processed by custom logic
+            if key == "tags":
+                for expectedTag in value:
+                    tagKeyExist = False
+                    tagKey = expectedTag.get("key")
+                    latestTags = state.get(key)
+                    for latestTag in latestTags:
+                        if ("key", tagKey) in latestTag.items():
+                            assert expectedTag == latestTag
+                            tagKeyExist = True
+                            break
+                    if tagKeyExist:
+                        continue
+                    assert expectedTag == None
+                continue
+            # parameterNameValues are processed by custom logic
+            if key == "parameterNameValues":
+                if not value:
+                    assert value == state.get(key)
+                    continue
+                for expectedParameterNameValue in value:
+                    parameterNameExist = False
+                    parameterName = expectedParameterNameValue.get("parameterName")
+                    latestParameterNameValues = state.get(key)
+                    for latestParameterNameValue in latestParameterNameValues:
+                        if ("parameterName", parameterName) in latestParameterNameValue.items():
+                            assert expectedParameterNameValue == latestParameterNameValue
+                            parameterNameExist = True
+                            break
+                    if parameterNameExist:
+                        continue
+                    assert expectedParameterNameValue == None
+                continue
+            assert value == state.get(key)
+
+    def assert_items_boto3(self, expectations: dict, state: dict) -> None:
+        if not expectations:
+            # nothing to assert as there are no expectations
+            return
+        if not state:
+            # there are expectations but no given state to validate
+            # following assert will fail and assert introspection will provide useful information for debugging
+            assert expectations == state
+
+        mdb = boto3.client("memorydb")
+        parameterGroupName = ""
+        for (latestKey, latestValue) in state.items():
+            if latestKey == "name":
+                parameterGroupName = latestValue
+        parameterGroups = mdb.describe_parameter_groups(ParameterGroupName=parameterGroupName)
+        parameterGroup = parameterGroups['ParameterGroups'][0]
+        latestTags = mdb.list_tags(ResourceArn=parameterGroup.get("ARN"))['TagList']
+        parameterNameValues = mdb.describe_parameters(ParameterGroupName=parameterGroupName)['Parameters']
```

Comment:
This should be moved into a method that each individual `ResourceHelper` overrides, then call it from within `ResourceHelper.assert_expectations` and pass the result as the `state` (please rename this to `actual`) into `assert_items_boto3`

Reply:
I don't find a way to override a part in for loop that calls continue/break there. Same as last comment.

Reply:
Mostly need to override entire `assert_items_boto3` function.

Reply:
Added a `get_aws_resource` function for it